assets/data_table/main.js

```diff
@@ -169,6 +194,21 @@ function App({ ctx, data }) {
     setColumns(newColumns);
   }, [colSizes]);
 
+  useEffect(() => {
+    const currentMenu = menu ? columns[menu.column].id : null;
+    const themeOverride = { bgHeader: "#F0F5F9" };
+    const newColumns = columns.map((header) => ({
+      ...header,
+      themeOverride: header.id === currentMenu ? themeOverride : null,
+    }));
+    setColumns(newColumns);
+  }, [showMenu]);
+
+  useEffect(() => {
+    const showMenu = menu ? true : false;
+    setShowMenu(showMenu);
+  }, [menu]);
```

Comment:
Can't we just determine it on the fly?

```suggestion
  const showMenu = !!menu;
```

Reply:
Yep, you're right. The problem I'm facing is that clicking on the header triggers the menu, but it also triggers the  `onOutsideClick` immediatly. So when I use a const I get` false -> true -> false` with only one click. The header is out of the menu bounds. I think it's possible what you're suggesting I just couldn't get there.

Reply:
Ok, Glide has two examples, one using a const and another using only one `useEffect`. The main difference is that in both it's not possible to close the menu from the menu button. Tbh, I prefer our behaviour. Mainly coz in our case you have to click outside the menu but inside the iframe to close it.

https://user-images.githubusercontent.com/5660941/204189636-9274d67f-2336-465d-96db-5b4cc49b8f8e.mov


https://user-images.githubusercontent.com/5660941/204189695-3f941181-9e8b-4e1b-b007-4535c0492c4b.mov

Reply:
> The problem I'm facing is that clicking on the header triggers the menu, but it also triggers the onOutsideClick immediatly

Ah I see, so it makes sense that deferring `showMenu` by one render cycle helps, let's do that for now!